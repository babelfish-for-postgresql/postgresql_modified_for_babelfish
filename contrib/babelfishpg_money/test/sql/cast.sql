SELECT CAST('2147483647'::FIXEDDECIMAL AS INT);

-- Ensure overflow is detected
SELECT CAST('2147483648'::FIXEDDECIMAL AS INT);

SELECT CAST('-2147483648'::FIXEDDECIMAL AS INT);

-- Ensure underflow is detected
SELECT CAST('-2147483649'::FIXEDDECIMAL AS INT);

SELECT CAST('32767'::FIXEDDECIMAL AS SMALLINT);

-- Ensure overflow is detected
SELECT CAST('32768'::FIXEDDECIMAL AS SMALLINT);

SELECT CAST('-32768'::FIXEDDECIMAL AS SMALLINT);

-- Ensure underflow is detected
SELECT CAST('-32769'::FIXEDDECIMAL AS SMALLINT);

SELECT CAST('1234321.23'::FIXEDDECIMAL AS FLOAT);

SELECT CAST('1234321.23'::FIXEDDECIMAL AS DOUBLE PRECISION);