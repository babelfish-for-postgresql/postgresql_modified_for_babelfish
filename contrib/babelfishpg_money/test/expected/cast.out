SELECT CAST('2147483647'::FIXEDDECIMAL AS INT);
    int4    
------------
 2147483647
(1 row)

-- Ensure overflow is detected
SELECT CAST('2147483648'::FIXEDDECIMAL AS INT);
ERROR:  integer out of range
SELECT CAST('-2147483648'::FIXEDDECIMAL AS INT);
    int4     
-------------
 -2147483648
(1 row)

-- Ensure underflow is detected
SELECT CAST('-2147483649'::FIXEDDECIMAL AS INT);
ERROR:  integer out of range
SELECT CAST('32767'::FIXEDDECIMAL AS SMALLINT);
 int2  
-------
 32767
(1 row)

-- Ensure overflow is detected
SELECT CAST('32768'::FIXEDDECIMAL AS SMALLINT);
ERROR:  smallint out of range
SELECT CAST('-32768'::FIXEDDECIMAL AS SMALLINT);
  int2  
--------
 -32768
(1 row)

-- Ensure underflow is detected
SELECT CAST('-32769'::FIXEDDECIMAL AS SMALLINT);
ERROR:  smallint out of range
SELECT CAST('1234321.23'::FIXEDDECIMAL AS FLOAT);
   float8   
------------
 1234321.23
(1 row)

SELECT CAST('1234321.23'::FIXEDDECIMAL AS DOUBLE PRECISION);
   float8   
------------
 1234321.23
(1 row)

